Fix flake8 linter warnings

edmorley opened the pull request:
Treeherder is switching to using peep. The peep docs suggest checking peep.py into the repo, but we run flake8 against our repo on Travis (flake8 = pyflakes+pep8, with some extra magic to make pyflakes output more usable). To avoid having to add peep to the flake8 ignore list, let's fix some of the warnings, since I imagine this will affect others too.

See the individual commit messages for the list of flake8 warnings the changes fix.

Thanks! :-)

erikrose commented:
Linting tools drive me crazy. :-) But thanks for the patch, Ed. It lowers a barrier to embedding peep, which is important. Since you're all tooled up, would you mind seeing if you can fix the 2 concerns about comments? Then we can merge it in. Thanks again!

erikrose commented:
@edmorley Not sure you got the above notifications, so...ping! :-)

edmorley commented:
I addressed those comments, but let me know if you'd prefer them done differently.

I also added a third commit that cleans up MismatchedReq.error(), both avoiding the indentation pattern that flake8 doesn't like, and (at least from my POV) makes it much more readable than it was previously :-)

edmorley commented:
And fwiw some of the linting rules do annoy me too, but at least for me, the advantage of running a linter all the time to catch the N % of real breakage cases (in the case of pyflakes) or to stop people introducing style deviations is generally worth the 5% of annoying cases :-)

erikrose commented:
Thanks, Ed!

edmorley commented:
Thanks! :-)
